File: doc-examples/example-java/src/main/java/example/Friend.java
package example;

import dev.langchain4j.service.SystemMessage;
import io.micronaut.langchain4j.annotation.RegisterAiService;

@RegisterAiService // <1>
public interface Friend {

    @SystemMessage("You are a good friend of mine. Answer using slang.") // <2>
    String chat(String userMessage);
}

File: test-suite-kotlin/src/main/kotlin/example/micronaut/Friend.kt
package example.micronaut

import dev.langchain4j.service.SystemMessage
import io.micronaut.langchain4j.annotation.AiService

@AiService
interface Friend {
    @SystemMessage("You are a good friend of mine. Answer using slang.")
    fun chat(userMessage: String): String
}
